DOC Update branch protection rules (#522)
silverstripe/developer-docs: en/12_Project_Governance/09_Repository_Management.md

If any exceptions to these rules need to be made for any reason, each exception must be approved in writing by the Silverstripe CMS Product Owner and only apply for a specified amount of time.

#### Branch protection rules

##### Updating branch and tag protection rulesets

> Use the `rulesets` command in [silverstripe/module-standardiser](https://github.com/silverstripe/module-standardiser) add and update the branch and tag protection rulesets for all support modules via the GitHub API.

Reviewer: ```suggestion
Use the `rulesets` command in [silverstripe/module-standardiser](https://github.com/silverstripe/module-standardiser) to add and update the branch and tag protection rulesets for all support modules via the GitHub API.
```

Author: Updated

These rulesets restrict non-admins from various activities such as merging pull-requests without a review, creating branches, and pushing tags.

##### Updating branch protection rule

A branch protection rule to protect against deleting branches is added by going to `https://github.com/silverstripe/<repository>/settings/branches`, clicking "Add rule", inputting `[0-9]*` as the branch name pattern, and clicking create (without checking any of the checkboxes).

> There is not a REST API endpoint available to add this rule as it uses a pattern, so it must be done manually.

Reviewer: ```suggestion
There is no REST API endpoint available to add this rule with a pattern, so it must be done manually.
```

Author: Updated
